### ISSUE TITLE
phpcbf does not report errors 

### ISSUE THREAD

**Issue by kkmuffme** [squizlabs/PHP_CodeSniffer#3642]
When a sniff has a PHP error, phpcbf just says "No fixable errors were found", whereas phpcs shows the sniff file + line + error message.

**Comment by jrfnl** [comment]
Too little information to go on. Also, not all errors are fixable.

**Comment by kkmuffme** [comment]
With PHP 8.1:

PHPCS says:

``` 
FILE: /some/file.php
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: trim(): Passing null to parameter #1 ($string) of type string is deprecated in
   |       | /my/directory/composer/vendor/wp-coding-standards/wpcs/WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php on line 280
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 471ms; Memory: 50.01MB
```

PHPCBF says:

```
No fixable errors were found
```

However with PHP 7.4 phpcbf does find and fix errors. phpcbf aborts due to the error in that sniff file. But without running phpcs you would never know about it.

**Comment by jrfnl** [comment]
@kkmuffme This is not an issue in PHP_CodeSniffer, but this is a known issue with WordPressCS and has already been fixed. Unfortunately, the release has been delayed for reasons. Until the new release has been tagged, run WPCS on PHP 7.4.

**Comment by jrfnl** [comment]
> We didn't realize that phpcbf wasn't running/auto-fixing for months, due to an error in a sniff that we don't have in our phpcs ruleset.

Just like `phpcs`, `phpcbf` will only run the sniffs which are in your ruleset, so this sounds unlikely.

All the same, your point about improving the user facing information when a PHP error occurs in one of the sniffs running fixers is well taken and something, which I agree, should be looked into.

**Comment by kkmuffme** [comment]
>Just like phpcs, phpcbf will only run the sniffs which are in your ruleset, so this sounds unlikely.

I think you misunderstood me:
phpcbf ran, but output `No fixable errors were found` even though there were fixable errors. phpcbf just didn't find any, because it aborted due to a PHP error in a sniff.
What I want to say is: if any sniff has a PHP error, phpcbf will report `No fixable errors were found`, which is wrong.
And this led to phpcbf not fixing anything for us, as it aborted before it could fix anything, due to a PHP error in a sniff.

**Comment by jrfnl** [comment]
@kkmuffme I didn't misunderstand you.

**Comment by gsherwood** [comment]
I get what you're saying and it's pretty easy to replicate undesired behaviour with a poorly coded sniff. PHPCBF does throw an exception of type `PHP_CodeSniffer\Exceptions\RuntimeException` like PHPCS does, but it's unhandled. For my version of PHP 8.1, that means the script ends with a fatal error and a non-zero exit code (255 in this case). I get the exact same behaviour on PHP 7.4.

I definitely don't get a message that no fixable errors were found; PHPCBF dies well before that.

I agree that improvement would be good here, but I feel like there might be a solution there somewhere with how the script is being run or your PHP settings. Is is possible that fatal errors are being suppressed in some way?

**Comment by rayshahriar** [comment]
> With PHP 8.1:
> 
> PHPCS says:
> 
> ```
> FILE: /some/file.php
> --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> FOUND 1 ERROR AFFECTING 1 LINE
> --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>  1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: trim(): Passing null to parameter #1 ($string) of type string is deprecated in
>    |       | /my/directory/composer/vendor/wp-coding-standards/wpcs/WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php on line 280
> --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> 
> Time: 471ms; Memory: 50.01MB
> ```
> 
> PHPCBF says:
> 
> ```
> No fixable errors were found
> ```
> 
> However with PHP 7.4 phpcbf does find and fix errors. phpcbf aborts due to the error in that sniff file, without any indication that it encountered an error. But without running phpcs you would never know about it.

I was getting a similar message. Downgrading to PHP 8.0 seemed to clear the issue for now.

**Comment by kkmuffme** [comment]
@gsherwood 
>PHPCBF does throw an exception of type

Where exactly? Could you send me a link to the specific line of the phpcbf sourcewhere that should happen in that case? bc I definitely doesn't happen for me for notices caused by PHP 8.1 specific issues.

>with how the script is being run

just basic CLI "php /.../phpcbf ..."

>Is is possible that fatal errors are being suppressed in some way?

Fatal errors cannot be suppressed to not stop code execution in PHP

**Comment by jrfnl** [comment]
> Fatal errors cannot be suppressed to not stop code execution in PHP

You may want to have a look at [set_error_handler()](https://www.php.net/manual/en/function.set-error-handler.php).... The fatals you are taking about are all catchable exceptions, so if PHPCS is run with a wrapper around it (like some IDEs do), this could well be happening.

**Comment by kkmuffme** [comment]
I think this was a misunderstanding: a "Fatal error" in my mind was an uncatchable exeption (or only catchable via register shutdown function last_error), like E_COMPILE_ERROR.

But @gsherwood you are right: we have a custom `set_exception_handler` that catches phpcbf errors (and logs them in sentry, which we didn't check for the CI pipeline...)
If that error would be handled in phpcbf like it is in phpcs, that would be great.

Issue is found though. Should I leave this open for that phpcbf change or close it?

**Comment by jrfnl** [comment]
Marking as duplicate of #2871

**Comment by alies-dev** [comment]
Guys, sorry, I'm confused. Are there any reasons why phpcbf does not provide a report (a list of unused errors) like phpcs does?

Make case
## phpcbf

```sh
phpcbf -p --standard=phpcs.xml --colors -d memory_limit=-1
....................................S.......................  60 / 176 (34%)
............................................................ 120 / 176 (68%)
........................................................     176 / 176 (100%)



No fixable errors were found

Time: 1.82 secs; Memory: 32MB
Script phpcbf -p --standard=phpcs.xml --colors -d memory_limit=-1 handling the phpcbf event returned with error code 2
```

## phpcs

```
phpcs -p -s --colors --report-full --report-summary --error-severity=1 --warning-severity=2 --cache=.cache/phpcs.cache -d memory_limit=-1
....................................S.......................  60 / 176 (34%)
...................................................E........ 120 / 176 (68%)
........................................................     176 / 176 (100%)



FILE: app/Console/Commands/RemindAboutMissingDailyUpdates.php
------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------
 30 | ERROR | Cognitive complexity for "handle" is 7 but has to be less than or equal to 5.
    |       | (SlevomatCodingStandard.Complexity.Cognitive.ComplexityTooHigh)
 30 | ERROR | Your function is too long. Currently using 44 lines. Can be up to 40 lines.
    |       | (SlevomatCodingStandard.Functions.FunctionLength.FunctionLength)
------------------------------------------------------------------------------------------------------------------------


PHP CODE SNIFFER REPORT SUMMARY
----------------------------------------------------------------------------------------------------------------------
FILE                                                                                                  ERRORS  WARNINGS
----------------------------------------------------------------------------------------------------------------------
app/Console/Commands/RemindAboutMissingDailyUpdates.php     2       0
----------------------------------------------------------------------------------------------------------------------
A TOTAL OF 2 ERRORS AND 0 WARNINGS WERE FOUND IN 1 FILE
----------------------------------------------------------------------------------------------------------------------

Time: 103ms; Memory: 22MB
Script phpcs -p -s --colors --report-full --report-summary --error-severity=1 --warning-severity=2 --cache=.cache/phpcs.cache -d memory_limit=-1 handling the phpcs event returned with error code 2
```
____

I found "No fixable errors were found" not very informative, so I need to run phpcbf and then phpcs to see these unfixed issues.

Version: v4.0
PHP 8.4
